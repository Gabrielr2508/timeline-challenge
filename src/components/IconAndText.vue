<template>
  <span class="d-flex align-items-center">
    <svgicon
      :name="icon"
      :title="alt"
      width="14px"
      class="light-gray"
    />
    <span class="ml-1">{{ text }}</span>
  </span>
</template>

<script>
import './icons';

export default {
  props: {
    icon: {
      type: String,
      required: true,
      default: '',
    },
    text: {
      type: String,
      required: true,
      default: '',
    },
    alt: {
      type: String,
      required: true,
      default: '',
    },
  },
};
</script>

<style>

</style>
