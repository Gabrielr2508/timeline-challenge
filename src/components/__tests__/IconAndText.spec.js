import { shallowMount, createLocalVue } from '@vue/test-utils';
import SvgIcon from 'vue-svgicon';

import IconAndText from '../IconAndText.vue';

const localVue = createLocalVue();
localVue.use(SvgIcon, {
  tagName: 'svgicon',
});

describe('IconAndText.vue', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      alt: 'dummy',
      icon: 'calendar',
      text: '11/11/1990',
    };

    wrapper = shallowMount(IconAndText, {
      propsData: { ...props },
      localVue,
    });
  });


  test('renders IconAndText correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
