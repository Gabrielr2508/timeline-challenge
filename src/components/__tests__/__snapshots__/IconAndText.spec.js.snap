// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAndText.vue renders IconAndText correctly 1`] = `
<span
  class="d-flex align-items-center"
>
  <svgicon-stub
    class="light-gray"
    fill="true"
    height=""
    name="calendar"
    title="dummy"
    width="14px"
  />
   
  <span
    class="ml-1"
  >
    11/11/1990
  </span>
</span>
`;
